### ISSUE TITLE
CA1508: false positive when when combining null-coalesce with using statement

### ISSUE THREAD

**Issue by innix** [dotnet/roslyn-analyzers#5161]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.301](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

I enabled all the analyzers on 1 of my projects and noticed this false positive:

```csharp
public static void Test(int? n)
{
    var ms = new MemoryStream();

    _ = n ?? 300;

    using (ms)
    {
        // ...
    }
}
```

### Steps To Reproduce

Compile the above program with the following properties set in your `.csproj` file:

```xml
<PropertyGroup>
  <EnableNETAnalyzers>true</EnableNETAnalyzers>
  <AnalysisMode>AllEnabledByDefault</AnalysisMode>
</PropertyGroup>
```

### Expected behavior

`using (ms)` should not be flagged as dead code. I'm not even performing a null check so I assume it's analyzing some compiler-generated code.

### Actual behavior

It gets flagged.

## Additional context

 - The false positive is **not** triggered by the typical ways the `using` statement is written:
    ```csharp
    
    // Fine.
    using (var ms = new MemoryStream())
    {
        // ...
    }
    
    // Also fine.
    using var ms = new MemoryStream();
    // ...
    ```

- You can move the null-coalesce line anywhere in the method and it still triggers the false positive on the `using` statement line.

**Comment by poke** [comment]
I have a similar example that also produces a false CA1508 in a using statement:

```csharp
public void Test()
{
    var stream = GetStream();

    if (stream == null)
        return;

    using (stream) // <- 1508 here
    {
    }
}

private Stream GetStream() => null;
```

If you comment out the `stream == null` check, the suggestion disappears because `stream` now could be `null` of course. But the using statement shouldn’t be affected by this in the first place.

**Comment by ryzngard** [comment]
I couldn't repro this in a test with latest SDK. Any idea if this still repros in SDK 6?

**Comment by innix** [comment]
@ryzngard I can confirm the bug is gone with latest SDK 6 version (`6.0.100-preview.6.21355.2`).
